metrics/firebase/functions/test/main_test.dart

```diff
@@ -450,6 +440,688 @@ void main() {
+        whenTaskDocuments().thenReturn([]);
+        when(newDocumentSnapshot.documentID).thenReturn(buildId);
+
+        await onBuildUpdatedHandler(change, null);
+
+        verify(
+          collectionReference.where(
+            'data.id',
+            isEqualTo: buildId,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "does not increment the successful builds duration if the updated build document snapshot's duration is null",
+      () async {
+        final newBuildJson = testDataGenerator.generateBuildJson(
+          buildStatus: BuildStatus.successful,
+          duration: null,
+        );
+
+        whenChangeBeforeData().thenReturn(oldBuildDocumentData);
+        whenChangeAfterData().thenReturn(DocumentData.fromMap(newBuildJson));
+        whenDocument().thenReturn(documentReference);
+        whenTaskDocuments().thenReturn([]);
+
+        await onBuildUpdatedHandler(change, null);
+
+        final successfulDurationIncrementMatcher =
+            documentFieldIncrementMatcher('successfulBuildsDuration', 0);
+
+        verify(
+          documentReference.setData(
+            argThat(successfulDurationIncrementMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "does not increment the successful builds duration if the updated build's status is not successful",
+      () async {
+        final newBuildJson = testDataGenerator.generateBuildJson(
+          buildStatus: BuildStatus.unknown,
+        );
+
+        whenChangeBeforeData().thenReturn(oldBuildDocumentData);
+        whenChangeAfterData().thenReturn(DocumentData.fromMap(newBuildJson));
+        whenDocument().thenReturn(documentReference);
+        whenTaskDocuments().thenReturn([]);
+
+        await onBuildUpdatedHandler(change, null);
+
+        final successfulDurationIncrementMatcher =
+            documentFieldIncrementMatcher('successfulBuildsDuration', 0);
+
+        verify(
+          documentReference.setData(
+            argThat(successfulDurationIncrementMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "updates a build days document with project id equals to the build document snapshot's project id",
+      () async {
+        whenChangeBeforeData().thenReturn(oldBuildDocumentData);
+        whenChangeAfterData().thenReturn(buildDocumentData);
+        whenDocument().thenReturn(documentReference);
+        whenTaskDocuments().thenReturn([]);
+
+        await onBuildUpdatedHandler(change, null);
+
+        final projectIdMatcher = predicate<DocumentData>(
+          (data) => data.getString('projectId') == projectId,
+        );
+
+        verify(
+          documentReference.setData(
+            argThat(projectIdMatcher),
+            any,
+          ),
+        ).called(1);
+      },
+    );
+
+    test(
+      "increments a build day document's successful field value if the updated build document snapshot's status is successful",
```

Comment:
```suggestion
      "increments a build day document's successful field value if the build status updated to successful",
```
What do you think? Please, apply this suggestion to all similar situations.